addons/point_of_sale/models/pos_session.py

```diff
@@ -308,19 +309,35 @@ def _validate_session(self):
             # Users without any accounting rights won't be able to create the journal entry. If this
             # case, switch to sudo for creation and posting.
             sudo = False
-            if (
-                not self.env['account.move'].check_access_rights('create', raise_exception=False)
-                and self.user_has_groups('point_of_sale.group_pos_user')
-            ):
-                sudo = True
-                self.sudo().with_company(self.company_id)._create_account_move()
-            else:
-                self.with_company(self.company_id)._create_account_move()
-            if self.move_id.line_ids:
-                # Set the uninvoiced orders' state to 'done'
-                self.env['pos.order'].search([('session_id', '=', self.id), ('state', '=', 'paid')]).write({'state': 'done'})
-            else:
-                self.move_id.unlink()
+            try:
+                if (
+                    not self.env['account.move'].check_access_rights('create', raise_exception=False)
+                    and self.user_has_groups('point_of_sale.group_pos_user')
+                ):
+                    sudo = True
```

Comment:
`sudo` is assigned but never used. Sad that the style runbot doesn't find these.